codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/generators/protocol/ProtocolGenerator.kt

```diff
@@ -43,8 +74,19 @@ interface ProtocolTraitImplGenerator {
  */
 open class ProtocolGenerator(
     codegenContext: CodegenContext,
+    /**
+     * `Protocol` contains all protocol specific information. Each smithy protocol, eg. RestJson, RestXml, etc. will
+     * have their own implementation of the protocol interface which defines how an input shape becomes and http::Request
+     * and an output shape is build from an http::Response.
+     */
     private val protocol: Protocol,
+    /** Operations generate a `make_operation(&config)` method to build a `smithy_http::Operation` that can be dispatched
```

Comment:
Nit: generally, the `/**` has the entire line to itself, but not a big deal if ktlint doesn't care.